src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,102 @@
+package duke;
+
+import duke.command.Command;
+import duke.exception.DukeException;
+import duke.util.Parser;
+import duke.util.Storage;
+import duke.util.TaskList;
+import duke.util.Ui;
+
+/**
+ * Represents a ChatBot.
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private boolean isExit = false;
```

Comment:
this `isExit` doesn't seem like its being used / the variable is being shadowed in `run()`. This makes it quite confusing to know whether the function-local `isExit` is being used or the object variable `isExit` is being used.